quantum/rgblight.c

```diff
@@ -57,59 +57,82 @@ rgblight_config_t rgblight_config;
 LED_TYPE led[RGBLED_NUM];
 bool rgblight_timer_enabled = false;
 
-void sethsv(uint16_t hue, uint8_t sat, uint8_t val, LED_TYPE *led1) {
-  uint8_t r = 0, g = 0, b = 0, base, color;
+#ifdef RGBW
+void convert_rgb_to_rgbw(uint8_t* r, uint8_t* g, uint8_t* b, uint8_t* w) {
+    // Determine lowest value in all three colors, put that into
+    // the white channel and then shift all colors by that amount
+    *w = MIN(*r, MIN(*g, *b));
+    *r = *r - *w;
+    *g = *g - *w;
+    *b = *b - *w;
```

Comment:
I personally prefer the more explicit way because it's easier to read and understand at a glance while the compiler most likely generates the same instructions from both. If the proposed change is the preferred one for the project though, I can apply it.

Reply:
Has been fixed with 0f457c1